### PR TITLE
add openai call + api-key in gitignore + error handling

### PR DESCRIPTION
**Pull request by Leda909**
I also chenged msg.reply -> to msg.chanel.send ---- bc "This creates a more generalized response that is not tied to a specific user. In channels with many different users, this approach can contribute to a more inclusive and community-oriented interaction."

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,39 +1,58 @@
...
+  if (msg.content === '!hello') {
+    //msg.reply('Hey!');
+	msg.channel.send('Halliioo');
+  }
+
+  // Connect to OpenAI with Error Handling
+  try {
+    const chatCompletion = await openai.chat.completions.create({
+	  model: 'gpt-3.5-turbo-1106',
+      messages: [{ role: 'user', content: 'Say this is a test' }],
```
Comment (halimahexe) [inline index.js:43]: Can you explain what this is doing?

File: index.js [hunk 2]
```diff
@@ -1,39 +1,58 @@
...
+      max_tokens: 10
```
Comment (halimahexe) [inline index.js:44]: I think `max_tokens` counts the tokens in the question and the answer, so 10 would be very limited. Can we change to a bigger number?

File: index.js [hunk 3]
```diff
@@ -1,39 +1,58 @@
...
+    });
 
+    // Console log the chatCompletion in JSON
+    console.log(JSON.stringify(chatCompletion, null, 2));
+    //msg.reply(chatCompletion.choices[0].message.content);
+	msg.channel.send(chatCompletion.choices[0].message.content);
```
Comment (halimahexe) [inline index.js:50]: I see what you mean about being more inclusive but I like that `msg.reply()` shows who the bot is replying to, to be able to track the conversation. In a big channel, won't it be confusing if conversations are happening around the bot and it's not replying to specific people?

File: index.js [hunk 4]
```diff
@@ -1,39 +1,58 @@
...
-	intents: [
-		GatewayIntentBits.Guilds, 
-		//GatewayIntentBits.GuildMembers,
-		GatewayIntentBits.GuildMessages, 
-		GatewayIntentBits.MessageContent
-	] 
+const client = new Client({
+  intents: [
+    GatewayIntentBits.Guilds,
+    //GatewayIntentBits.GuildMembers,
```
Comment (halimahexe) [inline index.js:12]: I think I've added the permissions to access `GuildMembers` now, so we could try it

**Review by halimahexe — COMMENTED** [review]
Looks good but I had a few questions!

File: index.js (hunk 3 above)
Comment (Leda909) [reply index.js:50]: You are right. we can change it back. actually I change to --->
const response = chatCompletion.choices[0].message.content;
console.log('OpenAI Response:', response);
msg.reply(response);

File: index.js (hunk 4 above)
Comment (Leda909) [reply index.js:12]: ok. I will check it.

File: index.js (hunk 1 above)
Comment (Leda909) [reply index.js:43]: - `chatCompletion` is a variable, which will contain the response from the OpenAI API 

- by using the `openai.chat.completions.create` method provided by the OpenAI Node.js library for interacting with the OpenAI GPT. It's an asynchronous function that returns a Promise.

- `await` is a keyword, used in an asynchronous function to wait for the completion of an asynchronous operation. In this case it is wait for the OpenAI API request to be completed. (`const response`)

- Within the function the messages array represents the conversation between the user and the model.
However, I would change it to : `messages: [{ role: 'user', content: msg.content }],`

- Here you can find [documentation to openai node js]( https://www.npmjs.com/package/openai?activeTab=readme)

File: index.js (hunk 2 above)
Comment (Leda909) [reply index.js:44]: yes. I only limited to 10 because the free version has no so much token per day.

File: index.js (hunk 1 above)
Comment (Leda909) [reply index.js:43]: So I changed to this:
image

Comment (halimahexe) [reply index.js:43]: Sorry for lack of clarity. I specifically meant the bit where the first message to openAI is from the user but now that you've changed the content, I understand. Thanks!

**Pull request merged**